src/test/java/seedu/hirehub/logic/commands/SearchApplicationCommandTest.java

```diff
@@ -0,0 +1,242 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.hirehub.model.Model;
+import seedu.hirehub.model.ModelManager;
+import seedu.hirehub.model.UserPrefs;
+import seedu.hirehub.model.application.Application;
+import seedu.hirehub.model.job.Job;
+import seedu.hirehub.model.person.Person;
+import seedu.hirehub.model.person.SearchPredicate;
+import seedu.hirehub.model.status.Status;
+import seedu.hirehub.testutil.ApplicationBuilder;
+import seedu.hirehub.testutil.JobBuilder;
+import seedu.hirehub.testutil.PersonBuilder;
+import seedu.hirehub.testutil.SearchApplicationPredicateBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code SearchApplicationCommand}.
+ */
+public class SearchApplicationCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalJobList(), new UserPrefs(),
+            getTypicalApplicationList());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), getTypicalJobList(), new UserPrefs(),
+            getTypicalApplicationList());
+
+    @Test
+    public void equals() {
+        SearchPredicate<Application> firstDescriptor =
+                new SearchApplicationPredicateBuilder(ALICE_BACK_END).build();
+        SearchPredicate<Application> secondDescriptor =
+                new SearchApplicationPredicateBuilder(BENSON_BACK_END).build();
+
+        SearchApplicationCommand searchApplicationFirstCommand = new SearchApplicationCommand(firstDescriptor);
+        SearchApplicationCommand searchApplicationSecondCommand = new SearchApplicationCommand(secondDescriptor);
+
+        // same object -> returns true
+        assertTrue(searchApplicationFirstCommand.equals(searchApplicationFirstCommand));
+
+        // different types -> returns false
+        assertFalse(searchApplicationFirstCommand.equals(10));
+        assertFalse(searchApplicationFirstCommand.equals("String"));
+
+        // null -> returns false
+        assertFalse(searchApplicationSecondCommand.equals(null));
+
+        // different Job Predicates -> returns false
+        assertFalse(searchApplicationFirstCommand.equals(searchApplicationSecondCommand));
+    }
+
+    @Test
+    public void execute_noApplicationFound() {
+        String expectedMessage = String.format(MESSAGE_APPLICATIONS_LISTED_OVERVIEW, 0);
+        SearchPredicate<Application> applicationDescriptor =
+                new SearchApplicationPredicateBuilder().withJob(OFF_CYCLE).build();
+
+        SearchApplicationCommand command = new SearchApplicationCommand(applicationDescriptor);
+        expectedModel.updateFilteredApplicationList(x -> applicationDescriptor.test(x));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredApplicationList());
+    }
+
+    /**
+     * Test search_application method for searching a matching job in an uniqueFilteredApplicationList
```

Comment:
Should this be "app"? Also appears in other test methods

Reply:
Thx, done